src/hotspot/cpu/x86/macroAssembler_x86_pow.cpp

```diff
@@ -2538,17 +2538,28 @@ void MacroAssembler::fast_pow(XMMRegister xmm0, XMMRegister xmm1, XMMRegister xm
   Label L_2TAG_PACKET_48_0_2, L_2TAG_PACKET_49_0_2, L_2TAG_PACKET_50_0_2, L_2TAG_PACKET_51_0_2;
   Label L_2TAG_PACKET_52_0_2, L_2TAG_PACKET_53_0_2, L_2TAG_PACKET_54_0_2, L_2TAG_PACKET_55_0_2;
   Label L_2TAG_PACKET_56_0_2, L_2TAG_PACKET_57_0_2, L_2TAG_PACKET_58_0_2, start;
+  Label L_NOT_DOUBLE2;
 
   assert_different_registers(tmp, eax, ecx, edx);
 
   address static_const_table_pow = (address)_static_const_table_pow;
+  static double DOUBLE2 = 2.0;
 
   bind(start);
   subl(rsp, 120);
   movl(Address(rsp, 64), tmp);
   lea(tmp, ExternalAddress(static_const_table_pow));
   movsd(xmm0, Address(rsp, 128));
   movsd(xmm1, Address(rsp, 136));
+
+  // Special case: pow(x, 2.0) => x * x
```

Comment:
64-bit code also handles `Special case: pow(x, 0.5) => sqrt(x)`. Don't we need to add that as well for consistency?

Reply:
> 64-bit code also handles `Special case: pow(x, 0.5) => sqrt(x)`. Don't we need to add that as well for consistency?

This is the bug fix for C2's optimization of pow(x, 2.0).

pow(x, 0.5) will be handled in JDK-8265940 .
Thanks.